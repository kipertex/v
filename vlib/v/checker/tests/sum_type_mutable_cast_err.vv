type Abc = int | string

struct Bar {
mut:
	a Abc
}

struct Foo {
	b Bar
}

struct Bar2 {
	a Abc
}

struct Foo2 {
	b Bar2
}

fn main() {
	mut x := Abc(0)
	if x is int {
		_ := x + 5
	}
	f := Foo{Bar{Abc(0)}}
	if f.b.a is int {
		_ := f.b.a + 5
	}
	mut f2 := Foo2{Bar2{Abc(0)}}
	if f2.b.a is int {
		_ := f.b.a + 5
	}
}
