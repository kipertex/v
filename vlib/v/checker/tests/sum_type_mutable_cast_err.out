vlib/v/checker/tests/sum_type_mutable_cast_err.vv:23:10: error: cannot use operator `+` with `Abc`
   21 |     mut x := Abc(0)
   22 |     if x is int {
   23 |         _ := x + 5
      |                ^
   24 |     }
   25 |     f := Foo{Bar{Abc(0)}}
vlib/v/checker/tests/sum_type_mutable_cast_err.vv:27:14: error: cannot use operator `+` with `Abc`
   25 |     f := Foo{Bar{Abc(0)}}
   26 |     if f.b.a is int {
   27 |         _ := f.b.a + 5
      |                    ^
   28 |     }
   29 |     mut f2 := Foo2{Bar2{Abc(0)}}
vlib/v/checker/tests/sum_type_mutable_cast_err.vv:31:14: error: cannot use operator `+` with `Abc`
   29 |     mut f2 := Foo2{Bar2{Abc(0)}}
   30 |     if f2.b.a is int {
   31 |         _ := f.b.a + 5
      |                    ^
   32 |     }
   33 | }